machine.js

```diff
@@ -95,10 +95,11 @@ const machineToThen = machine => function(ctx, ev) {
       this.child = interpret(machine, s => {
         this.onChange(s);
         if(s.machine.state.value.final) {
+          if (s.parent && Object.is(s.parent.machine, this.child.machine)) return;
```

Comment:
More info about [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is)